runtime/lua/vim/health/health.lua

```diff
@@ -1,8 +1,14 @@
 local M = {}
 local health = require('vim.health')
 
-local shell_error = function()
-  return vim.v.shell_error ~= 0
+---Run a system command and return ok and its stdout.
+---@param cmd string[]
+---@return boolean
+---@return string
+local function system(cmd)
+  local result = vim.system(cmd, { text = true }):wait()
+  -- NOTE: `or ''` to please the type checker
```

Comment:
don't need this comment. it's a common and idiomatic pattern (and the type checker is signaling a valid problem)

Reply:
👍🏼. The problem is not valid though: `result.stdout` is only nil when `vim.system` receives `stdout = function() end` in opts, which it is not. I'm not sure if that type definition can be improved (i.e. if luals handles those more fine-grained type annotations?)